src/compiler/checker.ts

```diff
@@ -5326,7 +5326,7 @@ namespace ts {
                     const typeId = type.id;
                     const isConstructorObject = getObjectFlags(type) & ObjectFlags.Anonymous && type.symbol && type.symbol.flags & SymbolFlags.Class;
                     const id = getObjectFlags(type) & ObjectFlags.Reference && (type as TypeReference).node ? "N" + getNodeId((type as TypeReference).node!) :

```

Comment:
No `isTypeReference`? Or did that cause problems? (Or, is this PR just not exhaustive?)

Reply:
It's just non-exhaustive.